contracts/Nitro-SCW.sol

```diff
@@ -47,17 +46,9 @@ contract NitroSmartContractWallet is IAccount {
         require(htlc.hashLock == keccak256(preImage), "Invalid preImage");
 
         removeActiveHTLC(hashLock);
-        updateLatestExpiry();
 
-        // Only tranfer funds if they're targetted outside of this wallet
-        if (htlc.to != address(this)) {
-            htlc.to.transfer(htlc.amount);
-        }
-
-        // If we've cleared the last HTLC we are finalized and can send the funds to the intermediary
-        if (getStatus() == WalletStatus.FINALIZED) {
-            // If we are finalized then we can just send the funds to the owner
-            intermediary.transfer(intermediaryBalance);
+        if (htlc.to == intermediary) {
```

Comment:
Slightly unrelated change but I think it makes things safer (and easier to understand).